packages/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsProxy.cpp

```diff
@@ -875,4 +887,85 @@ void LayoutAnimationsProxy::maybeUpdateWindowDimensions(
   }
 }
 
+#ifdef ANDROID
+/*
+ * It is possible that we may finish the layout animation before the native view
+ * is mounted. If the view wasn't mounted, we wouldn't be able to restore the
+ * opacity of the view. To fix this, we need to schedule a React update that
+ * will restore the view's opacity. This is safe because we'll use the same
+ * queue where React has already scheduled (but not yet executed) the view
+ * mounting, so the opacity update will execute after the view is mounted.
+ */
+void LayoutAnimationsProxy::restoreOpacityInCaseOfFlakyEnteringAnimation(
+    const LayoutAnimation &currentLayoutAnimation,
+    Tag tag,
+    SurfaceId surfaceId) const {
+  if (!currentLayoutAnimation.opacity.has_value()) {
+    // Animation was successfully finished, and the opacity was restored, so we
+    // don't need to do anything. Only the Entering animation has a set opacity
+    // value.
+    return;
+  }
+  const auto weakThis = weak_from_this();
+  const auto viewOpacity = currentLayoutAnimation.opacity.value();
+  jsInvoker_->invokeAsync([=](jsi::Runtime &runtime) {
+    const auto self = weakThis.lock();
+    if (!self) {
+      return;
+    }
+    self->uiManager_->getShadowTreeRegistry().visit(
+        surfaceId, [=](ShadowTree const &shadowTree) {
+          shadowTree.commit(
+              [=](RootShadowNode const &oldRootShadowNode) {
+                const auto self = weakThis.lock();
+                if (!self) {
+                  return cloneShadowTreeWithNewProps(oldRootShadowNode, {});
+                }
+                const auto &rootShadowNode =
+                    static_cast<const ShadowNode &>(oldRootShadowNode);
+                const auto *targetShadowNode =
+                    self->findInShadowTreeByTag(rootShadowNode, tag);
+                if (targetShadowNode == nullptr) {
+                  return cloneShadowTreeWithNewProps(oldRootShadowNode, {});
+                }
+                PropsMap propsMap;
+                propsMap[&targetShadowNode->getFamily()].emplace_back(
+                    folly::dynamic::object("opacity", viewOpacity));
+                return cloneShadowTreeWithNewProps(oldRootShadowNode, propsMap);
+              },
+              {});
+        });
+  });
+}
+
+const ShadowNode *LayoutAnimationsProxy::findInShadowTreeByTag(
+    const ShadowNode &node,
+    Tag tag) const {
+  if (node.getTag() == tag) {
+    return const_cast<const ShadowNode *>(&node);
+  }
+  for (auto &child : node.getChildren()) {
+    if (const auto result = findInShadowTreeByTag(child, tag)) {
+      return result.get();
+    }
+  }
+  return nullptr;
+}
```

Comment:
This function needs to traverse the whole shadow tree in the worst case scenario, how about using `getAncestors` instead?